source/extensions/filters/http/jwt_authn/authenticator.h

```diff
@@ -19,6 +19,8 @@ typedef std::unique_ptr<Authenticator> AuthenticatorPtr;
 
 typedef std::function<void(const ::google::jwt_verify::Status& status)> AuthenticatorCallback;
```

Comment:
Not sure about this func callback now. It made sense when it is just the status that needs to be returned, but now every new thing that needs to be returned will get a new func callback?

Reply:
Agreed that it is not ideal.   The best solution is to define a callback interface that has addPayload() and onComplete() call, and later can add call for new data.   I did try to explore this route,  it is not easy; which object should implement such interface,  maybe ContextImpl,   it is good for addPayload, but not good for onComplete() since OnComplete can be called per Verifier which there could be many, not just one.

So I settled down with two call_back functions for now,  it is easier to pass in two lambda functions.